enterprise/ha/src/test/java/org/neo4j/kernel/ha/TestBasicHaOperations.java

```diff
@@ -127,6 +131,115 @@ public void testBasicPropagationFromSlaveToMaster() throws Throwable
     }
 
     @Test
+    public void testBasicPropagationFromSlaveToMasterUsingCypher() throws Throwable
+    {
+        // given
+        clusterManager = new ClusterManager( clusterOfSize( 3 ), dir.cleanDirectory( "propagation" ),
+                stringMap( GraphDatabaseSettings.cache_type.name(), "hpc", HaSettings.tx_push_factor.name(), "1" ) );
+        clusterManager.start();
+        ClusterManager.ManagedCluster cluster = clusterManager.getDefaultCluster();
+
+        cluster.await( ClusterManager.allSeesAllAsAvailable() );
+
+        HighlyAvailableGraphDatabase master = cluster.getMaster();
+        HighlyAvailableGraphDatabase slave = cluster.getAnySlave();
+        HighlyAvailableGraphDatabase otherSlave = cluster.getAnySlave( slave );
+
+        try( Transaction tx = master.beginTx() )
+        {
+            master.execute( "CREATE (n:Test {name: \"test 1\"})" );
```

Comment:
I think you need to add a (test?) dependency on Cypher before this method works.